modules/fedimint-wallet-tests/tests/tests.rs

```diff
@@ -188,14 +197,57 @@ async fn on_chain_peg_in_and_peg_out_happy_case() -> anyhow::Result<()> {
         deposit_updates.next().await.unwrap(),
         DepositStateV2::WaitingForTransaction
     );
+
+    info!(?address, "Peg-in address generated");
+    let (_proof, tx) = bitcoin
+        .send_and_mine_block(
+            &address,
+            bsats(PEG_IN_AMOUNT_SATS)
+                + bsats(wallet_module.get_fee_consensus().peg_in_abs.msats / 1000),
+        )
+        .await;
+
+    info!("Waiting for confirmation");
     assert!(matches!(
         deposit_updates.next().await.unwrap(),
-        DepositStateV2::WaitingForConfirmation { .. }
+        DepositStateV2::WaitingForConfirmation { btc_out_point, .. } if btc_out_point.txid == tx.txid()
     ));
+
+    bitcoin.mine_blocks(finality_delay).await;
+
+    // Afaik technically not necessary, but useful to speed up test (should probably
```

Comment:
Hmm, I wouldn't have expected this to be necessary since we have a few checks if we're in a test and poll aggressively (1ms or 100ms depending on path).

I'm a bit suspicious of the behavior outside of a test environment now.

Reply:
Interesting, looks like `is_running_in_test_env` returns false when running locally with `cargo test -p fedimint-wallet-tests`

Reply:
I read through the code that determines wake up times and it's wayyyy over-engineered imo. There are two layers of wake up logic:

1. The outer one is told by the lowest of all `next_check_time` when to wake up next. There is also a `min_sleep` that responds to being in a test env, but that only sets a minimum wait before anything can happen, so doesn't help.
2. The inner layer is triggered by the outer and determines which addresses actually need checking. This one checks more often if it runs in the dev env, but the outer layer blocks us from getting there.

I think the logic could be massively simplified, it's not that hard of a problem and the complexity here seems accidental.